### PR TITLE
feat(api): dedicated /health liveness probe; repoint Docker HEALTHCHECK

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

Add a dedicated liveness healthcheck endpoint `GET /health` and repoint the Docker `HEALTHCHECK` at it, removing the container's reliance on the incidental root route `GET /`.

- New `HealthModule` / `HealthController`: `GET /health` returns `{ status: 'ok' }` with **no database or downstream dependency** (liveness, not readiness). Public (no guard) and not throttled, so an orchestrator can always reach it.
- `Dockerfile` `HEALTHCHECK` now probes `/health` (was `/`) and `--start-period` is raised `10s → 30s` for boot headroom — aligning with the `vev-smart-evse-adapter` pattern (dedicated liveness route + start period).
- No new dependency: `@nestjs/terminus` is intentionally **not** added — a pure liveness check is a trivial controller.

## Context

The previous `HEALTHCHECK` hit `GET /`, which only returns 2xx because an incidental `AppController.getHello()` ("Hello World!") exists. That couples container liveness to a boilerplate business route that could be removed or moved behind a prefix. A dedicated, dependency-free endpoint is a stable contract for liveness.

## Verification

- `npm run lint:check` (0 errors), `npm run build`, `npm test` (748 pass), `npm run test:e2e` (16 pass).
- `docker compose up -d --build` then `docker inspect --format '{{json .State.Health}}' brasse-bouillon-api` → `"Status":"healthy"` (FailingStreak 0); `GET /health` → `200 {"data":{"status":"ok"}}`.

## Checklist

- [x] Tests added — unit + e2e, Happy / Sad / Edge (200 + envelope, public/no-auth contract, `POST /health` → 404)
- [x] Lint + build + unit + e2e green
- [x] Container `HEALTHCHECK` verified healthy on a fresh build
- [x] No new dependency; named exports; no `any`; English-only artifacts

## Summary by CodeRabbit

* **New Features**
  * Added a `/health` endpoint that provides the API's operational status for monitoring and deployment health checks.

* **Chores**
  * Updated Docker container healthcheck to target the new health endpoint instead of the root path.

### CODE REVIEW

**Comment by benoit-bremaud** [comment]
Pour info : l'API dispose maintenant d'une adresse dédiée qui répond simplement « je suis en vie » (GET /health).

Avant, l'outil qui surveille si l'application tourne (le « healthcheck » de Docker, le système qui fait tourner l'appli en conteneur) interrogeait la page d'accueil de l'API. Ça fonctionnait un peu par hasard, parce qu'une page de test « Hello World » existait à cet endroit — si on l'avait retirée un jour, la surveillance aurait cru à tort que l'application était en panne. On a donc ajouté un point de contrôle dédié, très léger, qui ne touche pas à la base de données et confirme uniquement que le serveur répond bien. Docker pointe désormais sur cette adresse, et on a laissé un peu plus de temps au démarrage avant la première vérification.

Rien à faire de spécial de votre côté : c'est une amélioration de fiabilité côté déploiement, invisible pour les utilisateurs.

File: packages/api/src/health/health.controller.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+@ApiTags('Health')
+@Controller('health')
+export class HealthController {
+  @Get()
+  @ApiOperation({ summary: 'Liveness probe — 200 while the process is up' })
+  @ApiOkResponse({ type: HealthStatusDto })
+  live(): HealthStatusDto {
+    return { status: 'ok' };
+  }
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/health/health.controller.ts:29]: Thanks — I looked into this and I'm intentionally **not** adding `@Public()` here, for three repo-specific reasons:

1. **The decorator doesn't exist yet.** The only reference in the codebase is a forward-looking note in `scan.controller.ts` ("will switch to `@Public` once #718 lands"). Adding `@Public()` now would not compile.
2. **There is no global auth guard** (no `APP_GUARD`, no `useGlobalGuards` in `main.ts`/`app.module.ts`), so the metadata `@Public()` would set has nothing to read — it would be a no-op.
3. **The established precedent for a public endpoint here is `FeedbackController`**, which is intentionally unguarded by simply omitting `JwtAuthGuard` (see its class doc comment). `HealthController` follows that exact pattern.

If/when #718 introduces a global guard plus a real `@Public()` decorator, the public endpoints (feedback and health alike) will be migrated together. Keeping it consistent with the current codebase for now.

File: packages/api/src/health/health.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      controllers: [HealthController],
+    }).compile();
+
+    controller = module.get<HealthController>(HealthController);
+  });
+
+  // Happy path — the liveness handler returns the constant acknowledgement.
+  it('returns the ok status acknowledgement', () => {
+    expect(controller.live()).toEqual({ status: 'ok' });
+  });
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/health/health.controller.spec.ts:30]: Holding on this one. `live()` is a constant, dependency-free function that returns `{ status: 'ok' }` — it has no inputs to validate and no collaborators that can fail, so there is no meaningful **sad path** at the unit level. The only failure mode is the process being down, which is exactly what the container `HEALTHCHECK` detects, not something a unit test can assert.

This is called out in the spec's header comment as an intentional H/S/E exception. The edge / method-surface behaviour is covered in `test/health.e2e-spec.ts` (`POST /health` → 404). The local `pr-pre-reviewer` review reached the same conclusion (no sad path, by design).

File: packages/api/test/health.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    const res = await request(app.getHttpServer()).get('/health').expect(200);
+
+    const body = res.body as { data?: { status?: unknown } };
+    expect(body.data?.status).toBe('ok');
+  });
+
+  // Edge — only GET is exposed; POST /health is not a route → 404.
+  it('exposes only GET — POST /health is a 404', async () => {
+    await request(app.getHttpServer()).post('/health').expect(404);
+  });
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/test/health.e2e-spec.ts:55]: Same reasoning as the unit spec: a pure liveness probe has no sad path. It takes no request body (nothing to reject with 400) and touches no database or downstream dependency (nothing to fail with 5xx), so a "failure scenario" test would assert behaviour that does not exist.

The suite deliberately covers the **Happy path** (200 + `ok` through the response envelope, including the public / no-auth contract) and an **Edge case** (method surface: `POST /health` → 404). The intentional absence of a sad path is documented in the file header so it reads as a design decision, not a coverage gap.

**Comment by benoit-bremaud** [comment]
## Review summary — ready to merge

**Branch:** `feat/api-liveness-healthcheck` · **Head:** `065a2f7`

### CI
All required checks green: `api` (pass), `Build (validate)` (pass), `changes` (pass), `security-audit` (pass), `CodeRabbit` (pass). Other package jobs skipped by path filter (back-end-only change).

### Reviews
- **Copilot** — reviewed 7/7 files, **0 comments**.
- **CodeRabbit** (ASSERTIVE) — 3 comments, all answered inline and **resolved**:
  - `@Public()` decorator → declined: the decorator does not exist in the codebase yet (only a planned `#718` note in `scan.controller.ts`), there is no global auth guard for it to affect, and the public-endpoint precedent (`FeedbackController`) simply omits the guard — which this controller does.
  - Unit + e2e "sad path" → declined: a constant, dependency-free liveness probe has no sad path (no input to reject, no dependency to fail); the only failure mode is the process being down, which the container `HEALTHCHECK` detects. Documented as an intentional H/S/E exception; Happy + Edge (method surface) are covered.

### Verification
- `npm run lint:check` (0 errors), `npm run build`, `npm test` (748 pass), `npm run test:e2e` (16 pass).
- Container: `docker compose up -d --build` → `docker inspect … .State.Health` = `"Status":"healthy"` (FailingStreak 0); `GET /health` → `200 {"data":{"status":"ok"}}`.

No outstanding conversations. Ready to merge.
